drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -2734,6 +2734,7 @@ static void rswitch_fib_event_add(struct rswitch_fib_event_work *fib_work)
 	mutex_lock(&rdev->priv->ipv4_forward_lock);
 	list_add(&new_routing_list->list, &rdev->routing_list);
 	mutex_unlock(&rdev->priv->ipv4_forward_lock);
+	new_routing_list->nh = nh;
```

Comment:
I believe you need to fully initialize `new_routing_list` before calling `list_add` or else there can be other race conditions.

Reply:
@lorc Thank you for your review. I've pushed a new version of PR. Could you please check it?